### PR TITLE
config.obfuscate

### PR DESCRIPTION
**Pull request by abrichr**
Resolves https://github.com/MLDSAI/OpenAdapt/issues/309

Example output:

```
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - CACHE_DIR_PATH=.cache
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - CACHE_ENABLED=True
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - CACHE_VERBOSITY=0
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - DB_ECHO=False
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - DB_FNAME=openadapt.db
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - OPENAI_API_KEY=************************v>
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - OPENAI_MODEL_NAME=gpt-3.5-turbo
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - RECORD_READ_ACTIVE_ELEMENT_STATE=False
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - REPLAY_STRIP_ELEMENT_STATE=True                         
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - IGNORE_WARNINGS=False                                   
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - ACTION_TEXT_SEP=-
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - ACTION_TEXT_NAME_PREFIX=<
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - ACTION_TEXT_NAME_SUFFIX=>
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - SCRUB_ENABLED=True                                      
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - SCRUB_CHAR=*                                            
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - SCRUB_LANGUAGE=en
2023-06-25 10:21:04.187 | INFO     | openadapt.config:<module>:125 - SCRUB_FILL_COLOR=255
2023-06-25 10:21:04.188 | INFO     | openadapt.config:<module>:125 - SCRUB_CONFIG_TRF={'nlp_engine_name': 'spacy', 'models': [{'lang_code': 'en', 'model_name': 'en_core_web_trf'}]}
2023-06-25 10:21:04.188 | INFO     | openadapt.config:<module>:125 - SCRUB_IGNORE_ENTITIES=['DATE_TIME', 'URL']              
2023-06-25 10:21:04.188 | INFO     | openadapt.config:<module>:125 - SCRUB_KEYS_HTML=['text', 'canonical_text', 'title', 'state', 'task_description', 'key_char', 'canonical_key_char', 'key_vk', 'children']
2023-06-25 10:21:04.188 | INFO     | openadapt.config:<module>:125 - ROOT_DIRPATH=/Users/abrichr/oa/OpenAdapt
2023-06-25 10:21:04.188 | INFO     | openadapt.config:<module>:125 - DB_FPATH=/Users/abrichr/oa/OpenAdapt/openadapt.db       
2023-06-25 10:21:04.188 | INFO     | openadapt.config:<module>:125 - DB_URL=sqlite:////Users/abrichr/oa/OpenAdapt/openadapt.db
2023-06-25 10:21:04.188 | INFO     | openadapt.config:<module>:125 - DIRNAME_PERFORMANCE_PLOTS=performance
```

### CODE REVIEW

File: openadapt/config.py [hunk 1]
```diff
@@ -104,9 +104,24 @@ def getenv_fallback(var_name):
...
+    num_reveal = int(len(val) * pct_reveal)
+    num_obfuscate = len(val) - num_reveal
+    obfuscated = char * num_obfuscate
+    revealed = val[-num_reveal:]
+    rval = f"{obfuscated}{revealed}"
+    assert len(rval) == len(val), (val, rval)
+    return rval
+
+
+
```
Comment (KrishPatel13) [inline openadapt/config.py:118]: Should  we remove extra new line here? As we should have a 2 newlines after a function. Here 3 were found. ?

File: openadapt/config.py [hunk 2]
```diff
@@ -104,9 +104,24 @@ def getenv_fallback(var_name):
 DB_URL = f"sqlite:///{DB_FPATH}"
 DIRNAME_PERFORMANCE_PLOTS = "performance"
 
+
+def obfuscate(val, pct_reveal=0.1, char="*"):
```
Comment (KrishPatel13) [inline openadapt/config.py:108]: char = SCRUB_CHAR ? Does this sound better (reusing) ?

File: openadapt/scripts/scrub.py [hunk 3]
```diff
@@ -86,6 +87,12 @@ def scrub_mp4(
         Path to the scrubbed (redacted) mp4 file.
     """
 
+    logger.info(f"{mp4_file=}")
+    logger.info(f"{scrub_all_entities=}")
+    logger.info(f"{playback_speed_multiplier=}")
```
Comment (KrishPatel13) [inline openadapt/scripts/scrub.py:92]: 🚀

**Review by KrishPatel13 — COMMENTED** [review]
Reviewed!

**Comment by 0dm** [comment]
Works well, but there's one issue - if the user has not specified an OPENAI_API_KEY, we shouldn't obfuscate the "<set your api key in .env>" message: 
image

**Comment by abrichr** [comment]
Thanks @0dm ! Addressed in abac4e5

**Comment by 0dm** [comment]
lgtm!

**Pull request merged**